core/control-plane/control-plane-aggregate-services/src/main/java/org/eclipse/edc/connector/controlplane/services/query/QueryValidatorFactory.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2024 Zichuan
+ * 
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * SPDX-License-Identifier: Apache-2.0
+ * 
+ * Contributors:
+ *       Zichuan - initial API and implementation
+ */
+
+package org.eclipse.edc.connector.controlplane.services.query;
+
+/**
+ * Factory object that is used to generate a new instance of a QueryValidator specific to the service.
+ */
+@FunctionalInterface
+public interface QueryValidatorFactory {
```

Comment:
This interface is not needed since these factories are not used interchangeably.